slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbConnection.java

```diff
@@ -61,12 +77,40 @@ public static void configureJdbi(final Jdbi jdbi) {
   }
 
   private static HikariDataSource createDataSource(
-      final String jdbcUrl, final String username, final String password) {
-    final HikariDataSource dataSource = new HikariDataSource();
-    dataSource.setJdbcUrl(jdbcUrl);
-    dataSource.setUsername(username);
-    dataSource.setPassword(password);
-    dataSource.addDataSourceProperty(PG_SOCKET_TIMEOUT_PARAM, DEFAULT_PG_SOCKET_TIMEOUT_SECONDS);
-    return dataSource;
+      final String jdbcUrl,
+      final String username,
+      final String password,
+      final Path hikariConfigurationFile) {
+    final Properties hikariConfigurationProperties =
+        loadHikariConfigurationProperties(hikariConfigurationFile);
+
+    final HikariConfig hikariConfig = new HikariConfig(hikariConfigurationProperties);
+    hikariConfig.setJdbcUrl(jdbcUrl);
+    if (!isEmpty(username)) {
+      hikariConfig.setUsername(username);
+    }
+    if (!isEmpty(password)) {
+      hikariConfig.setPassword(password);
+    }
+
+    return new HikariDataSource(hikariConfig);
+  }
+
+  @VisibleForTesting
+  static Properties loadHikariConfigurationProperties(final Path hikariConfigurationFile) {
+    final Properties hikariConfigurationProperties = new Properties();
+    if (hikariConfigurationFile != null) {
+      try (FileInputStream inputStream = new FileInputStream(hikariConfigurationFile.toFile())) {
+        hikariConfigurationProperties.load(inputStream);
+      } catch (final IOException e) {
+        throw new UncheckedIOException(e);
```

Comment:
Perhaps add some text saying it can't load the hikari properties file?

Reply:
done.